File: frontend/src/types/order.ts
/**
 * Order types — maps to backend order API endpoints.
 */

// ---------------------------------------------------------------------------
// Order Item
// ---------------------------------------------------------------------------

export interface OrderItemInput {
  product_id: string;
  product_name: string;
  quantity: number;
  unit_price: number;
}

// ---------------------------------------------------------------------------
// Order Address
// ---------------------------------------------------------------------------

export interface OrderAddress {
  name: string;
  phone: string;
  address: string;
  landmark: string;
  city: string;
  state: string;
  pincode: string;
}

// ---------------------------------------------------------------------------
// Requests
// ---------------------------------------------------------------------------

export interface PlaceOrderRequest {
  user_id: string;
  cart_items: OrderItemInput[];
  address: OrderAddress;
  payment_method: PaymentMethod;
  total_amount: number;
}

// ---------------------------------------------------------------------------
// Responses
// ---------------------------------------------------------------------------

export interface OrderResponse {
  order_id: string;
  status: string;
  estimated_delivery: string;
  total_amount: number;
  payment_method: string;
  address: OrderAddress;
  items: OrderItemInput[];
  created_at: string;
}

export interface OrderListResponse {
  orders: OrderResponse[];
}

// ---------------------------------------------------------------------------
// Payment Methods
// ---------------------------------------------------------------------------

export type PaymentMethod =
  | "cod"
  | "upi"
  | "credit_card"
  | "debit_card"
  | "net_banking";

export const PAYMENT_METHODS: { value: PaymentMethod; label: string; icon: string }[] = [
  { value: "cod", label: "Cash on Delivery", icon: "💵" },
  { value: "upi", label: "UPI", icon: "📱" },
  { value: "credit_card", label: "Credit Card", icon: "💳" },
  { value: "debit_card", label: "Debit Card", icon: "💳" },
  { value: "net_banking", label: "Net Banking", icon: "🏦" },
];

File: backend/app/schemas/order.py
"""Order schemas — Pydantic models for order API endpoints."""

from __future__ import annotations

from uuid import UUID

from pydantic import BaseModel, Field


class OrderItemInput(BaseModel):
    """Single item in an order."""

    product_id: str
    product_name: str
    quantity: int = Field(ge=1)
    unit_price: float = Field(ge=0)


class OrderAddress(BaseModel):
    """Delivery address for an order."""

    name: str
    phone: str
    address: str
    landmark: str = ""
    city: str
    state: str
    pincode: str


class OrderCreateRequest(BaseModel):
    """Request body to place a new order."""

    user_id: UUID
    cart_items: list[OrderItemInput]
    address: OrderAddress
    payment_method: str  # "cod" | "upi" | "credit_card" | "debit_card" | "net_banking"
    total_amount: float = Field(ge=0)


class OrderResponse(BaseModel):
    """Single order response."""

    order_id: str
    status: str  # "confirmed"
    estimated_delivery: str
    total_amount: float
    payment_method: str
    address: OrderAddress
    items: list[OrderItemInput]
    created_at: str


class OrderListResponse(BaseModel):
    """List of orders for a user."""

    orders: list[OrderResponse]
